service/history/api/get_workflow_util.go

```diff
@@ -238,14 +238,29 @@ func GetMutableState(
 func MutableStateToGetResponse(
 	mutableState workflow.MutableState,
 ) (*historyservice.GetMutableStateResponse, error) {
-	currentBranchToken, err := mutableState.GetCurrentBranchToken()
-	if err != nil {
+	// NOTE: fields of GetMutableStateResponse (returned value) are accessed outside of workflow lock
+	// and, therefore, MUST be copied by value from mutableState fields.
+
+	var currentBranchToken []byte
+	if cbt, err := mutableState.GetCurrentBranchToken(); err != nil {
 		return nil, err
+	} else {
+		currentBranchToken = make([]byte, len(cbt))
+		copy(currentBranchToken, cbt)
 	}
 
 	executionInfo := mutableState.GetExecutionInfo()
 	workflowState, workflowStatus := mutableState.GetWorkflowStateStatus()
 	lastFirstEventID, lastFirstEventTxnID := mutableState.GetLastFirstEventIDTxnID()
+
+	var mostRecentWorkerVersionStamp *commonpb.WorkerVersionStamp
+	if mrwvs := mutableState.GetExecutionInfo().GetMostRecentWorkerVersionStamp(); mrwvs != nil {
+		mostRecentWorkerVersionStamp = &commonpb.WorkerVersionStamp{
+			BuildId:       mrwvs.GetBuildId(),
+			UseVersioning: mrwvs.GetUseVersioning(),
+		}
```

Comment:
```suggestion
		mostRecentWorkerVersionStamp := mutableState.GetExecutionInfo().GetMostRecentWorkerVersionStamp()
		if mostRecentWorkerVersionStamp != nil {
			mostRecentWorkerVersionStamp = common.CloneProto(mostRecentWorkerVersionStamp)
		}
```

Reply:
I don't think anyone does or will mutate the fields inside the stamp. But anyways, cloning is a good idea for extra safety.

Reply:
I'd still suggest using CloneProto to avoid missing any new fields